### DISCUSSION TITLE
Alterando um produto

### DISCUSSION THREAD

**Discussion by Igao02** [balta-io/forum#453] [Tira Dúvidas] [answered]
Boa noite Balta, e pessoal, espero que estejam bem.
Balta, na aula "Alterando um produto" eu não consegui executar o método async sendo void, quando eu clicava em "Editar", ele trava a aplicação com o Erro: 

"System.ObjectDisposedException: 'Cannot access a disposed context instance. A common cause of this error is disposing a context instance that was resolved from dependency injection and then later trying to use the same context instance elsewhere in your application. This may occur if you are calling 'Dispose' on the context instance, or wrapping it in a using statement. If you are using dependency injection, you should let the dependency injection container take care of disposing context instances.
Object name: 'ApplicationDbContext'.'

Para resolver eu deixei o meu método sem ser void:
  [Parameter]
  public int Id { get; set; }
  public Product Model { get; set; } = new();
  public IEnumerable<Category> Categories { get; set; } = Enumerable.Empty<Category>();

  protected override async Task OnInitializedAsync()
  {
      Model = await Context
          .Products
          .AsNoTracking()
          .FirstOrDefaultAsync(x => x.Id == Id) ?? new();

      Categories = await Context
          .Categories
          .AsNoTracking()
          .OrderBy(x => x.Title)
          .ToListAsync();
  }

Mas eu não sei o motivo de não ter funcionado com void, poderia me explicar por favor??

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @Igao02, como vai? Você tem o código com VOID ai? Não entendi o motivo também...

**Comment by Igao02** [comment]
Balta, o código com void era o mesmo que você disponibilizou no git hub

Obter o Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by andrebaltieri** [reply]
Estranho... teoricamente não era para ter diferença... não no código que escrevi... tanto que rodou no vídeo!
Não sei dizer o motivo de ter dado dispose no Context... pode ter sido outra coisa.

**Comment by Igao02** [comment]
Achei muito estranho mesmo Balta, eu até copiei o seu código direto, mas não funcionou mesmo assim, deu o mesmo erro.

Obter o Outlook para Android<https://aka.ms/AAb9ysg>
________________________________
